core-api/src/test/java/com/optimizely/ab/optimizelyjson/OptimizelyJSONExtendedTest.java

```diff
@@ -0,0 +1,148 @@
+/**
+ *
+ *    Copyright 2020, Optimizely and contributors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+package com.optimizely.ab.optimizelyjson;
+
+import com.optimizely.ab.config.parser.UnsupportedOperationException;
+import com.optimizely.ab.optimizelyjson.types.MD1;
+import com.optimizely.ab.optimizelyjson.types.MD2;
+import com.optimizely.ab.optimizelyjson.types.MD3;
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class OptimizelyJSONExtendedTest extends OptimizelyJSONCoreTest {
+
+    @Test
+    public void testGetValueNullKeyPath() throws UnsupportedOperationException {
+        OptimizelyJSON oj1 = new OptimizelyJSON(orgJson, getParser());
+
+        MD1 md1 = oj1.getValue(null, MD1.class);
+        assertNotNull(md1);
+        assertEquals(md1.k1, "v1");
+        assertEquals(md1.k2, true);
+        assertEquals(md1.k3.kk1, 1.2, 0.01);
+        assertEquals(md1.k3.kk2.kkk1, true);
+        assertEquals((Double)md1.k3.kk2.kkk4[0], 5.7, 0.01);
+        assertEquals(md1.k3.kk2.kkk4[2], "vvv4");
+
+        // verify previous getValue does not destroy the data
```

Comment:
the purpose of this test is to call getValue twice with non empty json path on the same object. That will check if we don't modify the object by traversing and returning just a subset of the object.